service/dps/storm/common/src/test/java/eu/europeana/cloud/service/dps/storm/utils/TaskStatusCheckerTest.java

```diff
@@ -54,13 +54,16 @@ public void testExecutionWithMultipleTasks() throws Exception {
             if (i < 3)
                 assertFalse(task2killedFlag);
             task1killedFlag = taskStatusChecker.hasKillFlag(TASK_ID);
-            task2killedFlag = taskStatusChecker.hasKillFlag(TASK_ID2);
+            if (i < 5)
+                task2killedFlag = taskStatusChecker.hasKillFlag(TASK_ID2);
             Thread.sleep(6000);
         }
-        verify(taskInfoDAO, times(4)).hasKillFlag(eq(TASK_ID));
-        verify(taskInfoDAO, times(3)).hasKillFlag(eq(TASK_ID2));
+        verify(taskInfoDAO, times(8)).hasKillFlag(eq(TASK_ID));
+        verify(taskInfoDAO, times(5)).hasKillFlag(eq(TASK_ID2));
         assertTrue(task1killedFlag);
         assertTrue(task2killedFlag);
+        Thread.sleep(20000);
```

Comment:
Sleep on tests is not recommended.
You can use awaitability, we using in metis-framework and it's quite simple to use. Check it out, it's quite cool!

Reply:
Indeed. However, there is no situation where I don't want to wait for the passing of those seconds. So wait suits me perfectly in here.